packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -390,13 +392,21 @@ export class GarbageCollector implements IGarbageCollector {
             // For new documents, GC has to be explicitly enabled via the gcAllowed flag in GC options.
             this.gcEnabled = gcOptions.gcAllowed === true;
             // Set the Session Expiry only if the flag is enabled or the test option is set.
-            if (this.mc.config.getBoolean(runSessionExpiry) && this.gcEnabled) {
+            if (this.mc.config.getBoolean(runSessionExpiryKey) && this.gcEnabled) {
                 this.sessionExpiryTimeoutMs = defaultSessionExpiryDurationMs;
             }
         }
 
         // If session expiry is enabled, we need to close the container when the timeout expires
-        if (this.sessionExpiryTimeoutMs !== undefined) {
+        if (this.sessionExpiryTimeoutMs !== undefined
+            && this.mc.config.getBoolean(disableSessionExpiryKey) !== true) {
```

Comment:
Now that I think about this some more, will the new documents' gc metadata be updated to no longer have session expiry? This is a good side effect, but one we have to consider.

Reply:
Hmmmmmmmmmmmmmmmmmmmmmm...

Since it doesn't clear `this.sessionExpiryTimeoutMs`, `ContainerRuntime.formMetadata` will still find the value and write it down.

So this setting (and `Fluid.GarbageCollection.TestOverride.SessionExpiryMs`) are both exclusively about the moment of setting the timeout.  Neither affects what's written in the file.

@agarwal-navin @tyler-cai-microsoft - I think this is ok, but do either of you think we want to be able to override the value written in the next summary, either with a different number or to undefined?

Reply:
I believe our intent for the `Fluid.GarbageCollection.TestOverride.SessionExpiryMs` feature flag is for testing purposes only. So, this should not affect the value that is written in summary. Basically, we can test a real document by setting this value to a small number but when we remove this feature flag, it should start using the 30 day timeout. So, we should probably not update `this.sessionExpiryTimeoutMs` in this case.

Maybe the same logic applies to `disableSessionExpiryKey`. This should not affect what is written in the summary as this is a temporary override while any issues are fixed. There is already a feature flag that can be used to not set it in the first place. So, if the intention is to not have this in summary, just don't set that feature flag.